src/com/csciOOD/Game.java

```diff
@@ -42,23 +42,44 @@ public void run(){
     }
 
     // TODO: mitigate death spiral and lag scenarios
+    // CPU saver and smoothing tips inspired by http://www.java-gaming.org/index.php?PHPSESSID=0l3msuh17hfkam8a8kbr904ck1&/topic,24220.0
     private void gameLoop() {
-        // Shoot for the Stars!
-        double fpsLimit = 120;
-        double updateInterval = 1000000000 / fpsLimit; // Should / could be set to something else. Lower? higher?
+        double hertz = 120.0;
+        double updateInterval = 1000000000 / hertz;
         double lastUpdateTime = System.nanoTime();
 
+        double fpsLimit = 60;
+        double renderInterval = 1000000000 / fpsLimit;
+        double lastRenderTime = System.nanoTime();
+
+        int updatesInRenderLimit = 5;
+
         while (gameRunning) {
-            double now = System.nanoTime();
             if (!isPaused) {
+                double now = System.nanoTime();
+                int updateCount = 0;
 
-                while(now - lastUpdateTime > updateInterval) {
+                while(now - lastUpdateTime > updateInterval && updateCount < updatesInRenderLimit) {
                     lastUpdateTime += updateInterval;
+                    updateCount++;
                     updateGame();
                 }
 
                 repaint();
+
+                //Yield until it has been at least the target time between renders. This saves the CPU from hogging.
+                while ( now - lastRenderTime < renderInterval && now - lastUpdateTime < updateInterval) {
+                    Thread.yield();
+
+                    try {Thread.sleep(1);} catch(Exception e) {}
+
+                    now = System.nanoTime();
+                }
+            } else {
+                lastUpdateTime = System.nanoTime();
+                lastRenderTime = System.nanoTime();
```

Comment:
This was the source of the "updating while paused" bug.  It wasn't actually updating while paused, it was catching up as the time difference grew while it was paused, so now the time properly resets whenever the game is paused.